core/lib/l1_contract_interface/src/i_executor/structures/commit_batch_info.rs

```diff
@@ -9,7 +12,22 @@ use crate::Tokenizable;
 
 /// Encoding for `CommitBatchInfo` from `IExecutor.sol`
 #[derive(Debug)]
-pub struct CommitBatchInfo<'a>(pub &'a L1BatchWithMetadata);
+pub struct CommitBatchInfo<'a> {
+    pub l1_batch_with_metadata: &'a L1BatchWithMetadata,
+    pub l1_batch_commit_data_generator: Arc<dyn L1BatchCommitDataGenerator>,
```

Comment:
I had thoughts about that. Another idea I had was, given that `l1_batch_commit_data_generator` is known by `CommitBatches`, maybe having the `CommitBatchInfo` struct doesn't make sense anymore and `Tokenize` implementation for `CommitBatches` could be:
```rust
impl Tokenize for CommitBatches {
    fn into_tokens(self) -> Vec<Token> {
        let stored_batch_info = StoredBatchInfo(&self.last_committed_l1_batch).into_token();
        let l1_batches_to_commit = self
            .l1_batches
            .iter()
            .map(|batch| self.l1_batch_commit_data_generator.clone())
            .collect();

        vec![stored_batch_info, Token::Array(l1_batches_to_commit)]
    }
}
```

Reply:
I agree with Roman that we shouldn't store reference here.
This crate defines an _interface_ to the smart contract, and given that we now have two interfaces, I'd imagine that we will have two structures here (e.g. `RollupCommitBatchInfo` and `ValidiumCommitBatchInfo`) that implement `Tokenize` according to the info stored inside.

Reply:
> that we now have two interfaces

Not sure it's the right place to discuss, but is the two interfaces desired here? Alternatively it could be the same interface but storage diff would be diffeent/emppty. Was this discussed anywhere?

Reply:
I thought that later on we'll need to provide some kind of _alternative_ pubdata, not just lack of it (e.g. pubdata hash or something). If this assumption is incorrect, then one interface is probably fine, but as soon as we meet X/Y pattern, I think having two concrete interfaces would work better.

Reply:
Makes sense @popzxc, but will still ping @StanislavBreadless if he has an opinion on this